File: src/main/java/io/github/deepanshut041/security/repositories/UserRepository.java
package io.github.deepanshut041.security.repositories;

import io.github.deepanshut041.security.entities.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

    User findByUsername(String username);
}

File: backend/src/main/kotlin/io/github/deepanshut041/sod/data/repository/UserRepository.kt
package io.github.deepanshut041.sod.data.repository

import io.github.deepanshut041.sod.data.entity.UserEntity
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository: MongoRepository<UserEntity, String> {
    fun findByEmail(email: String): Optional<UserEntity>
    fun existsByEmail(email: String): Boolean
}